apps/condo/domains/resident/tasks/notifyResidentsOnPayday.js

```diff
@@ -0,0 +1,123 @@
+        if (conflictingMessageTypes.find(type => type === message.type)) return true
+    }
+    return false
+}
+
+async function sendNotification (context, receipt) {
+    const serviceConsumers = await ServiceConsumer.getAll(context, {
+        accountNumber: get(receipt, 'account.number', null),
+        deletedAt: null,
+    })
+
+    for (const consumer of serviceConsumers) {
+        const resident = consumer.resident
+        const userId = resident.user.id
+        const lang = get(COUNTRIES, [get(receipt, 'organization.country', conf.DEFAULT_LOCALE), 'locale'], conf.DEFAULT_LOCALE)
+        const now = dayjs().subtract(1, 'month')
+
+        if (await hasConflictingPushes(context, userId)) {
+            logger.info({ msg: 'Today the User received push notifications with a conflicting type', userId })
+            continue
+        }
+        
+        const uniqKey = `${userId}_notification_on_payday`
+        await sendMessage(context, {
+            ...DV_SENDER,
+            to: { user: { id: userId } },
+            lang: conf.DEFAULT_LOCALE,
+            type: NEED_TO_PAY_RECEIPT_ON_PAYDAY_MESSAGE_TYPE,
+            meta: {
+                dv: 1,
+                data: {
+                    monthName: lang === RU_LOCALE ? now.locale('ru').format('MMMM') : now.format('MMMM'),
+                    billingReceiptId: { required: true },
+                    serviceConsumerId: consumer.id,
+                    residentId: resident.id,
+                    userId,
+                    toPayAmount: receipt.toPay,
+                },
+            },
+            uniqKey,
+        })
+    }
+
+}
+
+async function notifyResidentsOnPayday () {
+    const { keystone: context } = await getSchemaCtx('User')
+    const now = dayjs()
+    const currentPeriod = now.set('date', 1).format('YYYY-MM-DD')
+    const previousPeriod = now.set('date', 1).subtract(1, 'month').format('YYYY-MM-DD')
+    logger.info({ msg: 'Start proceeding', startAt: now.format() })
+
+    // get all BillingReceipts with positive toPay field
+    const allBillingReceipts = (await loadListByChunks({
+        context,
+        list: BillingReceipt,
+        chunkSize: 20,
+        where: {
+            period_in: [
+                currentPeriod,
+                previousPeriod,
+            ],
+            deletedAt: null,
+        },
+    })).filter(receipt => receipt.toPay > 0)
+
+    for (const receipt of allBillingReceipts) {
+        const organizationId = get(receipt, ['context', 'organization', 'id'])
+        const accountNumber = get(receipt, ['account', 'number'])
+        const toPay = get(receipt, ['toPay'])
+        const paid = await getPaymentsSum(
+            context,
+            organizationId,
+            accountNumber,
+            get(receipt, 'period', null),
+            get(receipt, ['recipient', 'bic'], null),
+            get(receipt, ['recipient', 'bankAccount'], null)
+        )
+
+        if (Number(paid) < Number(toPay)) {
```

Comment:
This way it looks simpler.
Maybe to cast to Number somewhere before this condition?

```suggestion
        if (paid < toPay) {
```